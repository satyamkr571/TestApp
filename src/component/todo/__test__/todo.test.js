import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../todo";
import renderer from "react-test-renderer";

test("test", () => {
  expect(true);
});
afterEach(() => {
  cleanup();
});
it("should render todo component", () => {
  const todos = { id: "1", title: "Watch Movie", completed: false };

  render(<Todo todo={todos} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Watch Movie");
});
it("should render todo component", () => {
  const todos = { id: "2", title: "Cook Food", completed: true };

  render(<Todo todo={todos} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Cook Food");
  expect(todoElement).not.toContainHTML("<strike>");
});
it("matches snapshot", () => {
  const todo = { id: "2", title: "Cook Food", completed: true };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
it("matches snapshot 2", () => {
  const todo = { id: "3", title: "Take Rest", completed: true };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  //console.log(tree);
  expect(tree).toMatchSnapshot();
});
