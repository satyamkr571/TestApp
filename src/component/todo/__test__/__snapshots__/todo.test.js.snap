// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="todo"
  data-testid="todo-2"
>
  <strike>
    hi 
    <h3>
      Cook Food
    </h3>
  </strike>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="todo"
  data-testid="todo-3"
>
  <strike>
    hi 
    <h3>
      Take Rest
    </h3>
  </strike>
</div>
`;
